lms/validation/__init__.py

```diff
@@ -0,0 +1,11 @@
+from lms.validation._exceptions import ValidationError
+from lms.validation._parser import parser
+from lms.validation._oauth import CANVAS_OAUTH_CALLBACK_ARGS
+
+
+__all__ = ("parser", "CANVAS_OAUTH_CALLBACK_ARGS", "ValidationError")
```

Comment:
`lms.validation` is explicitly specifying its public API in a PEP 8-compliant way. This is a new thing that other packages in lms and h don't do yet:

1. The `__init__.py` contains an `__all__` that lists the public API names. This helps tooling to discover the public API and also disambiguates when a name without a leading underscore is being imported into `__init__.py` in order to add it to the package's public API, vs when a name without a leading underscore is being imported into `__init__.py` for `__init__.py`'s internal usage.

2. All internal modules, functions etc are named with a leading underscore. E.g. `_handle_error()`, `_parser.py`, `_exceptions.py` and `_oauth.py`.

The idea is that other packages import the names from `__init__.py`, not from submodules or subpackages. For example `from lms.validation import CANVAS_OAUTH_CALLBACK_ARGS`. Even the tests import the names this way, so the `_foo.py` modules are not being imported anywhere outside of `lms/validation/`.

Reply:
Strictly I think PEP 8 requires _all_ modules to have an `__all__`, so `_exceptions.py` and `_oauth.py` would need it too, but the intended public API of those modules is very clear already without it